bigquery_etl/cli/generate.py

```diff
@@ -99,7 +99,22 @@ def generate_all(ctx, output_dir, target_project, ignore, use_cloud_function):
     """Run all SQL generators."""
     click.echo(f"Generating SQL content in {output_dir}.")
     click.echo(ROOT / SQL_GENERATORS_DIR)
-    for _, cmd in reversed(generate.commands.items()):
+
+    def generator_command_sort_key(command):
+        match command.name:
+            # Run `stable_views` after `glean_usage` because both update `dataset_metadata.yaml` files
+            # and we want the `stable_views` updates to take precedence.
+            case "glean_usage":
+                return (1, command.name)
+            case "stable_views":
+                return (2, command.name)
```

Comment:
This is based on [comments in #2594](https://github.com/mozilla/bigquery-etl/pull/2594#discussion_r779871871).

Reply:
I believe `for _, cmd in reversed(generate.commands.items()):` causes `stable_views` to be run before `glean_usage`. I'm not sure whether it makes a difference here though? I believe `glean_app_ping_views` might depended on the stable views at some point.

Reply:
OK, I've reversed the order in this PR so that `stable_views` runs before `glean_usage`, and now no `dataset_metadata.yaml` changes appear in the `sql.diff`.